### DISCUSSION TITLE
Validate Domain don´t validate

### DISCUSSION THREAD

**Discussion by leoborlot** [postalserver/postal#2607] [Help with using Postal] [answered]
## Describe the bug

Hi people.  Ive install the last release. All right with the instalations (I have 2 server running same version).
When I add a domain to check the DNS registers (SPF< DKIM, RPF), I click in button, the spin load run, but dont stop, ater 30 seconds return a red alert. 

When I check the web log, this is the lines:

[10] [2023-09-07T20:13:20.500] INFO -- : Started POST "/org/expressmei/servers/serexpressmei/domains/49ca1e1b-c886-4103-a6dd-0668bd42c787/check" for 179.217.19.80 at 2023-09-07 20:13:20 +0000
[10] [2023-09-07T20:13:20.501] INFO -- : Processing by DomainsController#check as JSON
[10] [2023-09-07T20:13:20.501] INFO -- :   Parameters: {"org_permalink"=>"expressmei", "server_id"=>"serexpressmei", "id"=>"49ca1e1b-c886-4103-a6dd-0668bd42c787"}
[10] [2023-09-07T20:18:08.960] INFO -- : Completed 200 OK in 288460ms (Views: 0.4ms | ActiveRecord: 12.0ms | Allocations: 6489)

## To Reproduce

1. Go to org/server/domain validation, add a newdomain
2. Set all register right
3. Click button '[Check my records are correct]'
4. See error how I texto over

## Expected behaviour

The register was OK, if I manually validate in DB, and send email, this work and recive 10/10 in mail tester

## Screenshots

<img width="538" alt="image" src="https://github.com/postalserver/postal/assets/20361821/dfbff2ad-ce27-4d89-91c4-791151bbd158">

![image](https://github.com/postalserver/postal/assets/20361821/b01880fd-b680-4cad-8dcf-6a6fc44ef81a)

## Environment details 

 - OS: UBUNTU
 - Browser: chrome
 - Version [Postal](https://postalserver.io/) 2.1.4.]
 - Type [e.g. desktop, mobile etc...]

**Comment by sinnbeck** [comment]
Does it work if you add it on the other server? I just added as a test to my test server and it works fine (but of course complains the records are wrong)

**Reply by leoborlot** [reply]
other server was same behavior...

**Reply by sinnbeck** [reply, marked as answer]
Did you try setting it to use external dns like @willpower232 suggested?

**Reply by leoborlot** [reply]
yes, that tip works!

**Comment by willpower232** [comment]
If your Postal is struggling to check DNS records then you could try changing this setting https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/config/postal.defaults.yml#L17

**Reply by leoborlot** [reply]
Will, did it work. Thanks. 
You know how explain (or where mayI find more about) whats was modifdied with this parameter?

**Reply by willpower232** [reply]
I believe that sometimes server providers strongly prefer you to use their local DNS resolvers rather than trying to determine things yourself so this setting forces Postal to look closer to where it is installed to get DNS queries answered. Something like that anyway.

**Comment by leoborlot** [comment]
Thanks guys. Problem solve.
